### PR TITLE
Add boundary to multipart header according RFC1341

### PR DESCRIPTION
**Pull request by mubat**
According [RFC1341](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html) request header should contain one requires parameter, "boundary".
With this error codeception module REST, that I uses, can correctly send files to our server that under development

### CODE REVIEW

**Comment by mubat** [comment]
Maybe should add ability to set manual boundary value?

File: src/Client.php [hunk 1]
```diff
@@ -9,18 +10,18 @@
...
- * @method ResponseInterface post(string|UriInterface $uri, array $options = [])
- * @method ResponseInterface patch(string|UriInterface $uri, array $options = [])
- * @method ResponseInterface delete(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface getAsync(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface headAsync(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface putAsync(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface postAsync(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface patchAsync(string|UriInterface $uri, array $options = [])
- * @method Promise\PromiseInterface deleteAsync(string|UriInterface $uri, array $options = [])
+ * @method ResponseInterface get(string | UriInterface $uri, array $options = [])
```
Comment (GrahamCampbell) [inline src/Client.php]: What's with all the unrelated changes?

Comment (mubat) [reply src/Client.php]: Revert changes. (IDE lives his life)

File: src/Client.php [hunk 2]
```diff
@@ -294,6 +294,8 @@ private function applyOptions(RequestInterface $request, array &$options)
             'set_headers' => [],
         ];
 
+        $contentTypeHeaderName = 'Content-Type';
```
Comment (GrahamCampbell) [inline src/Client.php]: Why?

**Comment by Nyholm** [comment]
Sorry for the long delay. Could you please rebase this PR?

**Comment by Nyholm** [comment]
I dont see the change here. The test would pass without this patch.

**Comment by Nyholm** [comment]
Closing this due the patch seams invalid and no response.
